File: back-end/src/main/java/com/capstone/backend/jpa/OrderItem.java
package com.capstone.backend.jpa;

import jakarta.persistence.Column;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.MapsId;
import jakarta.persistence.Table;

@Entity
@Table(name = "order_items")
public class OrderItem {

    @EmbeddedId
    private OrderItemId id;

    @ManyToOne
    @MapsId("orderId")
    @JoinColumn(name = "order_id")
    private Order order;

    @ManyToOne
    @MapsId("medicineId")
    @JoinColumn(name = "medicine_id")
    private Medicine medicine;

    @Column(name = "quantity", nullable = false)
    private Byte quantity;
    
    @Column(name = "cost", nullable = false)
    private Double cost;
    
    public OrderItem() {}

    public OrderItem(OrderItemId id, Order order, Medicine medicine, Byte quantity) {
		this.id = id;
		this.order = order;
		this.medicine = medicine;
		this.quantity = quantity;
		this.cost = medicine.getPrice() * quantity;
	}

	public OrderItemId getId() {
        return id;
    }

    public void setId(OrderItemId id) {
        this.id = id;
    }

    public Order getOrder() {
        return order;
    }

    public void setOrder(Order order) {
        this.order = order;
    }

    public Medicine getMedicine() {
        return medicine;
    }

    public void setMedicine(Medicine medicine) {
        this.medicine = medicine;
    }

    public Byte getQuantity() {
        return quantity;
    }

    public void setQuantity(Byte quantity) {
        this.quantity = quantity;
    }
    
    public Double getCost() {
    	return cost;
    }

	public void setCost(Double cost) {
		this.cost = cost;
	}
	
	public String log() {
		return "Item = " + medicine.getName() + " X " + quantity;
	}

	@Override
	public String toString() {
		return "OrderItem [OrderId=" + id.getOrderId() + ", MedicineId=" + id.getMedicineId() + ", order=" + order + ", medicine=" + medicine + ", quantity=" + quantity + "]";
	}
    
}

File: front-end/src/app/models/order-item.ts
import { Medicine } from "./medicine";
import { Order } from "./order";

export class OrderItem {
    private id: OrderItemId;
    private order: Order;
    private medicine: Medicine;
    private quantity: number;
    private cost: number;

    constructor(id?: OrderItemId, order?: Order, medicine?: Medicine, quantity?: number, cost?: number) {
        this.id = id || new OrderItemId();
        this.order = order || new Order();
        this.medicine = medicine || new Medicine();
        this.quantity = quantity || 0;
        this.cost = cost || 0;
    }

    get Id() {
        return this.id;
    }

    set Id(value: OrderItemId) {
        this.id = value;
    }

    get Order() {
        return this.order;
    }

    set Order(value: Order) {
        this.order = value;
    }

    get Medicine() {
        return this.medicine;
    }

    set Medicine(value: Medicine) {
        this.medicine = value;
    }

    get Quantity() {
        return this.quantity;
    }

    set Quantity(value: number) {
        this.quantity = value;
    }

    get Cost() {
        return this.cost;
    }

    set Cost(value: number) {
        this.cost = value;
    }

    toString(): string {
        return `OrderItem [id=${this.id}, quantity=${this.quantity}, cost=${this.cost}]`;
    }
}

export class OrderItemId {

    private orderId: number;
    private medicineId: number;

    constructor(orderId?: number, medicineId?: number) {
        this.orderId = orderId || 0;
        this.medicineId = medicineId || 0;
    }

    get OrderId() {
        return this.orderId;
    }

    set OrderId(value: number) {
        this.orderId = value;
    }

    get MedicineId() {
        return this.medicineId;
    }

    set MedicineId(value: number) {
        this.medicineId = value;
    }
}